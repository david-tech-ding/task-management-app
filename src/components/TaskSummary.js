import React from "react";
import "../styles/tasksummary.css";
import axios from "axios";

const TaskSummary = ({ title, tasks }) => {
  axios.defaults.baseURL = "https://api.trello.com/1";
  //   const boardName = { title };

  //   axios
  //     .post("/boards", { boardName })
  //     .then(console.log(boardName))
  //     .catch((err) => console.log(err));
  const highPriorityTasks = tasks.filter(
    (task) => task.priorityLevel === "High"
  );
  const mediumPriorityTasks = tasks.filter(
    (task) => task.priorityLevel === "Medium"
  );
  const lowPriorityTasks = tasks.filter((task) => task.priorityLevel === "Low");
  return (
    <div className="dashboard">
      <div className="task-summary">
        <h2 className="task-summary-heading">{title}</h2>
        <div className="high-priority-summary">
          <h3 className="priority-summary-heading">High Priority</h3>
          {highPriorityTasks.length}
        </div>
        <div className="medium-priority-summary">
          <h3 className="priority-summary-heading">Medium Priority</h3>
          {mediumPriorityTasks.length}
        </div>
        <div className="low-priority-summary">
          <h3 className="priority-summary-heading">Low Priority</h3>
          {lowPriorityTasks.length}
        </div>
      </div>
    </div>
  );
};

export default TaskSummary;
