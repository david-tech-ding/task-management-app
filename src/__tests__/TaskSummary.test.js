import React from "react";
import { render, screen } from "@testing-library/react";
import TaskSummary from "../components/TaskSummary";

describe("renders TaskSummary correctly", () => {
  it("renders correctly", () => {
    const validProps = {
      title: "test title",
      tasks: [],
    };
    const { asFragment } = render(
      <TaskSummary title={validProps.title} tasks={validProps.tasks} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe("contains correct elements", () => {
  it("contains 3 priority level summaries", () => {
    const validProps = {
      title: "test title",
      tasks: [],
    };

    render(<TaskSummary title={validProps.title} tasks={validProps.tasks} />);
    const priorityLevelSummaries = screen.queryAllByRole("heading", {
      level: 3,
    });

    expect(priorityLevelSummaries).toHaveLength(3);
  });

  it("has a counter for each priority level which changes based on tasks", () => {
    const validProps = {
      title: "test title",
      tasks: [{ priorityLevel: "High" }],
    };

    render(<TaskSummary title={validProps.title} tasks={validProps.tasks} />);
    const highPriorityCounter = screen.getByText(/1/);
    expect(highPriorityCounter).toBeInTheDocument();
  });
});
