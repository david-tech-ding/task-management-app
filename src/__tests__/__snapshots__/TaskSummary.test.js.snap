// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TaskSummary correctly renders correctly 1`] = `
<DocumentFragment>
  <div
    class="dashboard"
  >
    <div
      class="task-summary"
    >
      <h2
        class="task-summary-heading"
      >
        test title
      </h2>
      <div
        class="high-priority-summary"
      >
        <h3
          class="priority-summary-heading"
        >
          High Priority
        </h3>
        0
      </div>
      <div
        class="medium-priority-summary"
      >
        <h3
          class="priority-summary-heading"
        >
          Medium Priority
        </h3>
        0
      </div>
      <div
        class="low-priority-summary"
      >
        <h3
          class="priority-summary-heading"
        >
          Low Priority
        </h3>
        0
      </div>
    </div>
  </div>
</DocumentFragment>
`;
